### ISSUE TITLE
Cropped text inside flowgraph rectangles

### ISSUE THREAD

**Issue by yassirnajmaoui** [huterguier/obsidian-mehrmaid#3]
It seems like every text I put inside quotes is cropped by one character (more or less).

Example:
Code:
```mehrmaid
---
title: Test chart
---
flowchart
    A["Some text"]
    B["Some text with an equation $Ax+b=y^{2n}$"]
	
	A -- "This $2x$ cropped too" ---> B
```
Output image:
![image](https://github.com/huterguier/obsidian-mehrmaid/assets/43155068/211ac8f7-3031-466a-8691-00d6bb786177)

It seems like it affects equations more than regular text.

**Comment by yassirnajmaoui** [comment]
It seems like the problem occurs vertically aswell

**Comment by huterguier** [comment]
Okay, that's weird. The example you sent seems to work for me:
![Screenshot from 2024-04-16 15-26-14](https://github.com/huterguier/obsidian-mehrmaid/assets/47152172/39898303-ad62-4f44-b136-964b112add58)
Did you install the plugin from the Obsidian plugin store?

Vertical cropping should not occur either, except for edge labels. But this is a problem of the Mermaid rendering/layout engine and i can't really do anything about it inside the scope of this plugin. If you encounter any other example where things get cropped just let me know.

**Comment by yassirnajmaoui** [comment]
Yes I did download it from the Obsidian plugin store directly from the Obsidian UI. Note that I am running on Linux in Ubuntu and Gnome, nothing fancy.

I also have the following problems:
- If I have two diagrams in the same document, when I switch the current view to "reading" instead of "editing", then switch back, the whole document gets broken and becomes uneditable. I have to reload it. I have the same issue when there is a compilation error in the me(h)rmaid code while in the "live preview" mode.
- If I export to PDF and I have two diagrams in the same document, the two diagrams overlap one another and the document becomes unintelligible.

I don't have any of these problems with "vanilla" mermaid.

Other than that I really enjoy the plugin. I just want to be able to write math formulas inside mermaid diagrams. I wish I had the Javascript skills to help you on this.

**Comment by vsiisv** [comment]
I have the same problem.
<img width="497" alt="image" src="https://github.com/huterguier/obsidian-mehrmaid/assets/129242125/e6147926-5f12-4af4-9af6-97c496f9d965">

I installed the plugin from GitHub.

**Comment by huterguier** [comment]
There should be no more issues when switching between reading and edit mode now. The reason was that both views render their own graphs, which lead to non-unique graph ids.
If you still encounter any sort of cropping, let me know and maybe give me some more detailed information on when this occurs.

**Comment by yassirnajmaoui** [comment]
Indeed, the Edit and Read mode bug seems to be fixed for me. But I still see the some cropping issue, but it did get better for the nodes at least (I still see it a lot for the text between the nodes, eg: `A -- "The text here" --> B`).

**Comment by yassirnajmaoui** [comment]
Also, I just realized that this bug doesn't happen when the following parameter is ON:
![image](https://github.com/huterguier/obsidian-mehrmaid/assets/43155068/ded0a80f-0d57-452a-bcae-3c6195828439)

**Comment by pirasleandro** [comment]
For me it's still broken. I have tried both the obsidian plugin store version and vanilla one. I tried to render the example from the store page, but it never looks like the screenshot from the store page. *(ignoring the broken image links)*

**Writing Mode:**
![image](https://github.com/huterguier/obsidian-mehrmaid/assets/69905493/4930dab8-e20d-4118-9e20-3fc517de2b7e)

**Reading Mode:**
![image](https://github.com/huterguier/obsidian-mehrmaid/assets/69905493/39535f66-a875-4668-8772-4a8903c17cc9)

**Screenshot from store page:**
![image](https://raw.githubusercontent.com/huterguier/obsidian-mehrmaid/HEAD/assets/example1.png)

**Comment by jforejtn** [comment]
I'm using a CSS hack to get rid of the text cropping. This is what I've added to my snippets:

```
.mehrmaid-markdown-container {
  padding: 0 8px 0 0;
}
```
Before:
![before](https://github.com/user-attachments/assets/b9d9c521-cd14-475b-9f3a-b608842c11ed)

After:
![after8](https://github.com/user-attachments/assets/8e8b3c63-9b6c-4d6a-b1ff-63197a4663d2)

**Comment by Francokab** [comment]
After fiddling a bit with it, I think the problem is that the font-size used to compute the height and width to use, and the font-size used when actually showing the text are different. The latter seems to be somewhat constant and set at 16, but the former can vary using the font size settings in obsidian and "Quick font size asjustement" hotkey. There is also a mismatch of font used.

Those discrepancy means that the size that are used to set the size of the rectangles are different from the ones that end up being actually needed.

To fix this, the renderMarkdown function should try and render the makdown using the exact same formating as when it will be put inside the meirmaid diagramm. I don't know how to do that as I have very little knowledge on using javascript and HTML/CSS but I hope that my findings will help someone to fix this issue.

**Comment by Francokab** [comment]
After some more looking, I found this temporary solution using this CSS snipet:

```
.mehrmaid-markdown-container {
    font-family: "trebuchet ms",verdana,arial,sans-serif;
    font-size: 16px;
    fill: #000000;
}
```

You can change the font-size and font-family to your liking as it will apply to both step of the rendering

Before:
<img alt="Image" src="https://github.com/user-attachments/assets/6475563a-fb9d-4e3b-9f83-11d8645265e2" />

After:
<img alt="Image" src="https://github.com/user-attachments/assets/6ae63161-9786-4a04-8b6c-8696d2d63603" />
